src/EditorFeatures/Core/Implementation/Suggestions/SuggestedAction.cs

```diff
@@ -168,14 +210,19 @@ void IDisposable.Dispose()
             // do nothing
         }
 
-        public virtual IEnumerable<SuggestedActionSet> ActionSets
+        public virtual bool HasActionSets
         {
             get
             {
-                return null;
+                return false;
             }
         }
 
+        public virtual Task<IEnumerable<SuggestedActionSet>> GetActionSetsAsync(CancellationToken cancellationToken)
+        {
+            return SpecializedTasks.Default<IEnumerable<SuggestedActionSet>>();
```

Comment:
SpecialiedTasks.EmptyEnumerable ... seems better or create SpecializedTasks.NullTask<T>() that cache and return Task.FromResult<T>(null);